lib/libraries/c3/extensions/paginating-legend.js

```diff
@@ -0,0 +1,217 @@
+var d3 = require('d3');
+
+var isDateString = require('../../../utils/assert-date-string');
+
+module.exports = function(cols){
+  var self = this, chart, columns, domNode, pagination, range;
+
+  chart = this.view._artifacts.c3;
+  columns = [];
+  domNode = this.el().querySelector('.' + this.theme() + '-rendering');
+  range = Math.round((domNode.offsetHeight - 78) / 20);
+  pagination = this.view._artifacts.pagination = {
+    hidden: [],
+    labels: [],
+    position: 0,
+    range: range,
+    total: 0
+  };
+
+  // Reduce threshold for donut/pie charts
+  if (this.type() === 'donut' || this.type() === 'pie') {
+    pagination.range = pagination.range >= 5 ? 5 : pagination.range;
+  }
+
+  // Build a reversed column list
+  for (var i = 0; i < cols.length; i++) {
+    if (cols[i][0] !== 'x' && !isDateString(cols[i][1])) {
+      columns.push(cols[i][0]);
+    }
+  }
+
+  var legendEl = d3.select(domNode)
+    .append('svg')
+    .attr('class', 'keen-c3-legend')
+    .attr('width', 120)
+    .attr('height', domNode.offsetHeight - 10)
+    .style('right', '-120px')
+    .style('top', '10px');
+
+  var legendItems = legendEl
+    .append('g')
+    .attr('class', 'keen-c3-legend-items');
+
+  var paginateElOffset = 20 * pagination.range;
+  var paginateEl = legendEl
+    .append('g')
+    .attr('class', 'keen-c3-legend-pagination')
+    .attr('transform', function(){
+      return 'translate(2, ' + paginateElOffset + ')'
+    });
+
+  paginateData();
+
+  function paginateData(){
+    pagination.labels = columns.slice(pagination.position, pagination.position + pagination.range);
+    pagination.total = columns.length;
+    renderLegendComponent.call(self, pagination.labels);
+    if (pagination.total > pagination.range) {
+      renderPaginationComponent.call(self);
+    }
+    chart.resize();
+  }
+
+  function renderLegendComponent(){
+    legendItems
+      .selectAll('g')
+      .remove();
+
+    var legendItemList = legendItems
+      .selectAll('g')
+      .data(pagination.labels);
+
+    legendItemList.enter()
+        .append('g')
+        .attr('transform', function(id, i){
+          return 'translate(0, ' + (20 * i) + ')'
+        })
+      .attr('data-id', function (id) { return id; })
+      .style('opacity', function(id){
+        var isHidden = pagination.hidden.indexOf(id);
+        return isHidden < 0 ? 1 : .35;
+      })
+      .each(function (id) {
+        d3.select(this)
+          .append('text')
+          .attr('font-size', 12)
+          .attr('pointer-events', 'none')
+          .attr('x', 15)
+          .attr('y', 9)
+          .text(id)
+          .text(function(id){
```

Comment:
Yep, this gives the text element content so `d3.select(this).node().getBBox().width` has a value work with when deciding whether or not to truncate the text.